server/etcdserver/apply/apply_kv_test.go

```diff
@@ -0,0 +1,415 @@
+// Copyright 2023 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apply
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap/zaptest"
+	"golang.org/x/crypto/bcrypt"
+
+	pb "go.etcd.io/etcd/api/v3/etcdserverpb"
+	"go.etcd.io/etcd/server/v3/auth"
+	"go.etcd.io/etcd/server/v3/etcdserver/api/membership"
+	"go.etcd.io/etcd/server/v3/etcdserver/api/v3alarm"
+	"go.etcd.io/etcd/server/v3/etcdserver/cindex"
+	"go.etcd.io/etcd/server/v3/etcdserver/errors"
+	"go.etcd.io/etcd/server/v3/lease"
+	serverstorage "go.etcd.io/etcd/server/v3/storage"
+	"go.etcd.io/etcd/server/v3/storage/backend"
+	betesting "go.etcd.io/etcd/server/v3/storage/backend/testing"
+	"go.etcd.io/etcd/server/v3/storage/mvcc"
+	"go.etcd.io/etcd/server/v3/storage/schema"
+)
+
+const (
+	putInherentCost = 256
+	key2            = "key2"
+	key3            = "key3"
+	key4            = "key4"
+	value64         = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789?!"
+	value128        = value64 + value64
+)
+
+var (
+	fourKeys = []pb.PutRequest{
+		{Key: []byte(key), Value: []byte("key's value")},
+		{Key: []byte(key2), Value: []byte("key2's value")},
+		{Key: []byte(key3), Value: []byte("key3's value")},
+		{Key: []byte(key4), Value: []byte("key4's value")},
+	}
+	updatedValue  = []byte("up2date value")
+	updatedValue2 = []byte("up2date value2")
+	zero          = []byte{0}
+)
+
+func putKeys(ctx context.Context, t *testing.T, keys []pb.PutRequest, app *applierV3backend) {
+	var prevRev int64
+	for _, k := range keys {
+		resp, _, err := app.Put(ctx, nil, &k)
+		require.NoError(t, err)
+		if prevRev > 0 {
+			require.Greater(t, resp.Header.Revision, prevRev)
+			prevRev = resp.Header.Revision
+		}
+	}
+}
+
+func defaultApplierV3Backend(t *testing.T) (*applierV3backend, backend.Backend) {
+	lg := zaptest.NewLogger(t)
+	be, _ := betesting.NewDefaultTmpBackend(t)
+	t.Cleanup(func() { be = nil })
```

Comment:
https://github.com/etcd-io/etcd/blob/af07f18dce96d4418bc1e89710675a44e3e5028b/server/storage/mvcc/kv_test.go#L758-L761

You can checkout this. We should call the `Close` to tear down the background goroutines.